MeshLib/MeshGenerators/RasterToMesh.cpp

```diff
@@ -40,208 +44,194 @@ MeshLib::Mesh* RasterToMesh::convert(
+        ((elem_type == MeshElemType::PRISM) || (elem_type == MeshElemType::HEXAHEDRON)))
+    {
+        ERR("Elevation mapping can only be performed for 2D meshes.");
         return nullptr;
+    }
 
-    MeshLib::Properties properties;
-    if (intensity_type == MeshLib::UseIntensityAs::MATERIALS)
+    MathLib::Point3d mesh_origin({header.origin[0] - (header.cell_size / 2.0),
+                                  header.origin[1] - (header.cell_size / 2.0),
+                                  header.origin[2]});
+    std::unique_ptr<MeshLib::Mesh> mesh (nullptr);
+    if (elem_type == MeshElemType::TRIANGLE)
+        mesh.reset(
+            MeshLib::MeshGenerator::generateRegularTriMesh(header.n_cols,
+                                                           header.n_rows,
+                                                           header.cell_size,
+                                                           mesh_origin,
+                                                           "RasterDataMesh"));
+    else if (elem_type == MeshElemType::QUAD)
+        mesh.reset(
+            MeshLib::MeshGenerator::generateRegularQuadMesh(header.n_cols,
+                                                            header.n_rows,
+                                                            header.cell_size,
+                                                            mesh_origin,
+                                                            "RasterDataMesh"));
+    else if (elem_type == MeshElemType::PRISM)
+        mesh.reset(
+            MeshLib::MeshGenerator::generateRegularPrismMesh(header.n_cols,
+                                                             header.n_rows,
+                                                             header.n_depth,
+                                                             header.cell_size,
+                                                             mesh_origin,
+                                                             "RasterDataMesh"));
+    else if (elem_type == MeshElemType::HEXAHEDRON)
+        mesh.reset(
+            MeshLib::MeshGenerator::generateRegularHexMesh(header.n_cols,
+                                                           header.n_rows,
+                                                           header.n_depth,
+                                                           header.cell_size,
+                                                           mesh_origin,
+                                                           "RasterDataMesh"));
+
+    MeshLib::Mesh* new_mesh(nullptr);
+    std::vector<std::size_t> elements_to_remove;
+    if (intensity_type == UseIntensityAs::ELEVATION)
     {
-        auto* const prop_vec = properties.createNewPropertyVector<int>(
-            "MaterialIDs", MeshLib::MeshItemType::Cell, 1);
-        fillPropertyVector<int>(*prop_vec, pix_val, pix_vis, header.n_rows, header.n_cols, elem_type);
+        std::vector<MeshLib::Node*> nodes(mesh->getNodes());
```

Comment:
not sure if the copy is needed here or if a (const) reference `std::vector<Node*> const& nodes = mesh->getNodes()` would be sufficient.

Reply:
Technically a reference would be okay but I'm editing coordinates, so I cannot use const&. And there is no non-const method that returns a reference.

Reply:
As far as I see there is no need to have a copy here: [this](https://github.com/ufz/ogs/commit/3346e29b935c31ea6455424cb8b9a63e54156405) compiles just fine.

Reply:
you are right, it does compile. although I dont' know why because I *am* changing the nodes...